### DISCUSSION TITLE
Progress bar

### DISCUSSION THREAD

**Discussion by lily2brain** [jspsych/jsPsych#1794] [Q&A] [answered]
Hi, I managed to add the progress bar to the experiment, but it is automatic, it advances by blocks, and since the last block is long, it advances only when it ends. How could you modify this and insert a percentage in the bar? I await an answer :)

**Comment by becky-gilbert** [comment, marked as answer]
Hi @lily2brain, check out the `jsPsych.setProgressBar` function: https://www.jspsych.org/core_library/jspsych-core/#jspsychsetprogressbar

**Reply by lily2brain** [reply]
Hi, thank you very much for the answer, I managed to do this by inserting this function in each block that wanted to progress in the bar: 
`on_finish: function(){
        jsPsych.setProgressBar(0.50); //insert percentage
      }
    }`
Thanks ;)
